site/src/pages/components/beforefocus.explainer.mdx

```diff
@@ -0,0 +1,128 @@
+---
+menu: Active Proposals
+name: beforefocus/focusNext (Explainer)
+layout: ../../layouts/ComponentLayout.astro
+---
+
+Authors: [Marat Abdullin](https://github.com/mshoho)
+
+{/* START doctoc generated TOC please keep comment here to allow auto update */}
+{/* DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE */}
+{/* END doctoc generated TOC please keep comment here to allow auto update */}
+
+## Introduction
+
+The browsers give very little options for implementing complex keyboard navigation
+patterns. We are currently stuck with workaround techniques like roving tabindex and/or
+dummy focusable elements to redirect keyboard focus. These techniques requre modifying the
+DOM on the fly and have limitations. They cause reflows and style recomputations, and
+could be hard or impossible to apply in some particular scenarios. For example, you might
+be owning a parent list component, but not owning the contents of the list items. Or
+you have to embed a cross-origin iframe you are unable to poke inside.
+
+While having additional new high level abstractions to handle keyboard navigation (like,
+for example, [focusgroup](./focusgroup.explainer)) sounds plausible, in reality it
+would be hard or impossible to align on the APIs that would cover all necessary cases —
+there is a multitude of different applications and no single standardized approach for them.
+It would also involve a huge body of work in the browser engine.
+
+Instead of high level APIs, this proposal focuses on having two fairly simple lower level
+functions which arguably would allow the application developers to implement whatever patterns
+they prefer without working around the limitations and impossibilities.
+
+Namely, one method called `document.focusNext()` and one event called `beforefocus`.
+Let's dive deeper into it.
+
+## document.focusNext()
+
+`document.focusNext()` method just exposes the default action the browsers have for `Tab` and
+`Shift+Tab` presses.
+
+The method's signature should be like:
+
+```ts
+focusNext(contextElement?: HTMLElement, direction?: FORWARD | BACKWARD);
```

Comment:
Does it feel weird to do `focusNext(undefined, "BACKWARD")`?

Reply:
Well, it's just an initial draft. It could either an object for options, or it could be two methods (`focusNext()` and `focusPrev()`). I suppose an object would be more future-proof in case some more options would be useful in the future.